### PR TITLE
V1.4_DG updates

### PR DESCRIPTION
**Pull request by PokezardVGC**
Added sequence diagrams and activity diagrams for:
- delete deadline 
- add deadline 
- help 
- edit

Added updates to DG

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -362,6 +428,45 @@ is trying to find. Here we have made an assumption that there the StudentId uniq
...
     * Cons: No clear distinction between tags and project status
 
+#### Implementation details
+
+The edit student feature is facilitated by `EditCommandParser` and `EditCommand`. The operation is exposed in the `Model` interface as `Model#Edit()`.
+
+Given below is an example usage scenario and how the edit student mechanism behaves at each step:
+
+1. The user enters delete deadline command and provides the student ID and other parameters(eg: student name, project name) to be changed to.
+2. `FYPManagerParser` creates a new `EditCommandParser` after preliminary processing of user input.
```
Comment (RussellDash332) [inline docs/DeveloperGuide.md:438]: FypManagerParser

**Review by RussellDash332 — COMMENTED** [review]
Small changes

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (PokezardVGC) [reply docs/DeveloperGuide.md:438]: On it boss

**Review by RussellDash332 — APPROVED** [review]
LGTM!

**Pull request merged**